com.intellij.diagnostic.PluginException: Cannot create extension (class=nl.hannahsten.texifyidea.index.file.LatexExternalEnvironmentIndex) [Plugin: nl.rubensten.texifyidea]

### Type of JetBrains IDE (IntelliJ, PyCharm, etc.) and version
PyCharm 2024.3.1.1 (build PY-243.22562.220)

### Operating System 
Linux 6.12.7-arch1-1 (amd64)

### TeXiFy IDEA version
0.9.9

### Description



### Stacktrace
```
com.intellij.diagnostic.PluginException: Cannot create extension (class=nl.hannahsten.texifyidea.index.file.LatexExternalEnvironmentIndex) [Plugin: nl.rubensten.texifyidea]
	at com.intellij.serviceContainer.ComponentManagerImpl.createError(ComponentManagerImpl.kt:989)
	at com.intellij.openapi.extensions.impl.XmlExtensionAdapter.doCreateInstance(XmlExtensionAdapter.kt:73)
	at com.intellij.openapi.extensions.impl.XmlExtensionAdapter.createInstance(XmlExtensionAdapter.kt:33)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.processAdapter(ExtensionPointImpl.kt:404)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.createExtensionInstances(ExtensionPointImpl.kt:377)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.getExtensionList(ExtensionPointImpl.kt:223)
	at com.intellij.openapi.extensions.ExtensionPointName.getExtensionList(ExtensionPointName.kt:54)
	at com.intellij.indexing.shared.platform.api.IdeSharedIndexesState.<init>(IdeSharedIndexesState.java:38)
	at com.intellij.indexing.shared.platform.api.SharedIndexInfrastructureVersion.getIdeVersion(SharedIndexInfrastructureVersion.java:68)
	at com.intellij.indexing.shared.platform.impl.SharedIndexStorageUtil.openFileBasedIndexChunks(SharedIndexStorageUtil.java:54)
        (...)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: com.intellij.diagnostic.PluginException: Cannot create class nl.hannahsten.texifyidea.index.file.LatexExternalEnvironmentIndex (classloader=PluginClassLoader(plugin=PluginDescriptor(name=TeXiFy-IDEA, id=nl.rubensten.texifyidea, descriptorPath=plugin.xml, path=~/.local/share/JetBrains/PyCharm2024.3/TeXiFy-IDEA, version=0.9.9, package=null, isBundled=false), packagePrefix=null, state=active, parents=PluginDescriptor(name=Grazie Lite, id=tanvd.grazi, moduleName=intellij.grazie.yaml, desc
	at com.intellij.serviceContainer.ComponentManagerImpl.doInstantiateClass(ComponentManagerImpl.kt:927)
        (...)
	... 76 more
Caused by: java.lang.NoClassDefFoundError: Could not initialize class nl.hannahsten.texifyidea.util.magic.CommandMagic
	at nl.hannahsten.texifyidea.index.file.LatexExternalEnvironmentDataIndexer.<init>(LatexExternalEnvironmentDataIndexer.kt:23)
	at nl.hannahsten.texifyidea.index.file.LatexExternalEnvironmentIndex.<init>(LatexExternalEnvironmentIndex.kt:19)
	at com.intellij.serviceContainer.ComponentManagerImpl.findConstructorAndInstantiateClass(ComponentManagerImpl.kt:909)
        (...)
	... 79 more
Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.NoClassDefFoundError: Could not initialize class nl.hannahsten.texifyidea.lang.commands.LatexGenericRegularCommand [in thread "DefaultDispatcher-worker-10"]
	at nl.hannahsten.texifyidea.util.magic.CommandMagic.<clinit>(CommandMagic.kt:39)
	at nl.hannahsten.texifyidea.index.file.LatexExternalEnvironmentDataIndexer.<init>(LatexExternalEnvironmentDataIndexer.kt:23)
	at nl.hannahsten.texifyidea.index.file.LatexExternalEnvironmentIndex.<init>(LatexExternalEnvironmentIndex.kt:19)
	at com.intellij.serviceContainer.ComponentManagerImpl.findConstructorAndInstantiateClass(ComponentManagerImpl.kt:909)
```

--- a/src/nl/hannahsten/texifyidea/index/file/LatexExternalEnvironmentDataIndexer.kt
+++ b/src/nl/hannahsten/texifyidea/index/file/LatexExternalEnvironmentDataIndexer.kt
@@ -20,7 +20,9 @@ class LatexExternalEnvironmentDataIndexer : DataIndexer<String, String, FileCont
         """(?=\\DescribeEnv\{?(?<key>[a-zA-Z_:]+\*?)}?\s*(?<value>[\s\S]{0,500}))""".toRegex()
 
     // e.g. \newenvironment{proof}{...} \newenvironment|\newtheorem|\NewDocumentEnvironment|\DeclareDocumentEnvironment|\ProvideDocumentEnvironment
-    private val directDefinitionRegex = """(${CommandMagic.environmentDefinitions.joinToString("|").replace("\\", "\\\\")})\*?\{(?<key>[a-zA-Z_:]+)}(?<value>)""".toRegex()
+    private val directDefinitionRegex by lazy {
+        """(${CommandMagic.environmentDefinitions.joinToString("|").replace("\\", "\\\\")})\*?\{(?<key>[a-zA-Z_:]+)}(?<value>)""".toRegex()
+    }
 
     override fun map(inputData: FileContent): MutableMap<String, String> {
         val map = mutableMapOf<String, String>()

--- a/src/nl/hannahsten/texifyidea/index/file/LatexExternalEnvironmentIndex.kt
+++ b/src/nl/hannahsten/texifyidea/index/file/LatexExternalEnvironmentIndex.kt
@@ -16,7 +16,7 @@ class LatexExternalEnvironmentIndex : FileBasedIndexExtension<String, String>() 
         val id = ID.create<String, String>("nl.hannahsten.texifyidea.external.environments")
     }
 
-    private val indexer = LatexExternalEnvironmentDataIndexer()
+    private val indexer by lazy { LatexExternalEnvironmentDataIndexer () }
 
     override fun getName(): ID<String, String> {
         return Cache.id

--- a/src/nl/hannahsten/texifyidea/index/file/LatexExternalEnvironmentIndex.kt
+++ b/src/nl/hannahsten/texifyidea/index/file/LatexExternalEnvironmentIndex.kt
@@ -16,7 +16,9 @@ class LatexExternalEnvironmentIndex : FileBasedIndexExtension<String, String>() 
         val id = ID.create<String, String>("nl.hannahsten.texifyidea.external.environments")
     }
 
-    private val indexer by lazy { LatexExternalEnvironmentDataIndexer () }
+    private val indexer by lazy {
+        LatexExternalEnvironmentDataIndexer()
+    }
 
     override fun getName(): ID<String, String> {
         return Cache.id

--- a/src/nl/hannahsten/texifyidea/util/magic/FileMagic.kt
+++ b/src/nl/hannahsten/texifyidea/util/magic/FileMagic.kt
@@ -12,21 +12,25 @@ object FileMagic {
      */
     val includeExtensions = listOf("sty", "tex", "cls", "bib")
 
-    val automaticExtensions = mapOf(
-        INCLUDE.cmd to LatexFileType.defaultExtension,
-        BIBLIOGRAPHY.cmd to BibtexFileType.defaultExtension
-    )
+    val automaticExtensions by lazy {
+        mapOf(
+            INCLUDE.cmd to LatexFileType.defaultExtension,
+            BIBLIOGRAPHY.cmd to BibtexFileType.defaultExtension
+        )
+    }
 
     /**
      * All possible file types.
      */
-    val fileTypes = setOf(
-        LatexFileType,
-        StyleFileType,
-        ClassFileType,
-        BibtexFileType,
-        TikzFileType
-    )
+    val fileTypes by lazy {
+        setOf(
+            LatexFileType,
+            StyleFileType,
+            ClassFileType,
+            BibtexFileType,
+            TikzFileType
+        )
+    }
 
     /**
      * All file extensions that have to be deleted when clearing auxiliary files.

